docs/user/openstack/install_upi.md

```diff
@@ -0,0 +1,871 @@
+         Serial Number:
+         Organization:   system:nodes
+Events:  <none>
+```
+
+If it does (this one is for `openshift-qlvwv-worker-0` which we've created earlier), you can approve it:
+
+```sh
+$ oc adm certificate approve csr-88jp8
+```
+
+Approved nodes should now show up in `oc get nodes`, but they will be in the `NotReady` state. They will create a second CSR which you should also review:
+
+```sh
+$ oc get csr -A
+NAME        AGE     REQUESTOR                                                                   CONDITION
+csr-2scwb   17m     system:serviceaccount:openshift-machine-config-operator:node-bootstrapper   Approved,Issued
+csr-5jwqf   17m     system:node:openshift-qlvwv-master-0                                         Approved,Issued
+csr-7mv4d   13s     system:node:openshift-qlvwv-worker-1                                         Pending
+csr-88jp8   3m29s   system:serviceaccount:openshift-machine-config-operator:node-bootstrapper   Approved,Issued
+csr-9dt8f   17m     system:node:openshift-qlvwv-master-1                                         Approved,Issued
+csr-bqkw5   18m     system:serviceaccount:openshift-machine-config-operator:node-bootstrapper   Approved,Issued
+csr-bx7p4   28s     system:node:openshift-qlvwv-worker-0                                         Pending
+csr-dpprd   99s     system:serviceaccount:openshift-machine-config-operator:node-bootstrapper   Approved,Issued
+csr-dtcws   117s    system:serviceaccount:openshift-machine-config-operator:node-bootstrapper   Approved,Issued
+csr-lj7f9   17m     system:node:openshift-qlvwv-master-2                                         Approved,Issued
+csr-lrtlk   17m     system:serviceaccount:openshift-machine-config-operator:node-bootstrapper   Approved,Issued
+csr-wkm94   18m     system:serviceaccount:openshift-machine-config-operator:node-bootstrapper   Approved,Issued
+csr-wqpfd   21s     system:node:openshift-qlvwv-worker-2                                         Pending
+```
+
+(we see the CSR approved earlier as well as a new `Pending` one for the same node: `openshift-qlvwv-worker-0`)
+
+And approve:
+
+```sh
+$ oc adm certificate approve csr-bx7p4
+```
+
+Once this CSR is approved, the node should switch to `Ready` and pods will be scheduled on it.
+
+### Wait for the OpenShift Installation to Complete
+
+Run the following command to verify the OpenShift cluster is fully deployed:
+
+```sh
+$ openshift-install --log-level debug wait-for install-complete
+```
+
+Upon success, it will print the URL to the OpenShift Console (the web UI) as well as admin username and password to log in.
+
+
+## Destroy the OpenShift Cluster
+
+First, remove the `api` and `*.apps` DNS records.
+
+Then run the following commands to delete all the OpenStack resources we've created:
+
+```sh
+$ openstack floating ip delete 203.0.113.23 203.0.113.19
+$ openstack server list --column Name --format value | grep "$INFRA_ID" | xargs openstack server delete
```

Comment:
Could you add `--no-run-if-empty` to xargs so that it doesn't fail when there is no matching server?